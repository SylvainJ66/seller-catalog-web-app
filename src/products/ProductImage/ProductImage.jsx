import { string } from "prop-types";
export default function ProductImage({ name, imageUrl }) {
  return <img src={imageUrl} alt={name} />;
}

ProductImage.propTypes = {
  imageUrl: string,
  name: string,
};
