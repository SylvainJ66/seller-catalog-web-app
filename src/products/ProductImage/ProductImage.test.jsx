import { render, screen } from "@testing-library/react";
import ProductImage from "@/products/ProductImage/ProductImage.jsx";

describe("<ProductImage />", () => {
  test("renders the image", () => {
    render(<ProductImage imageUrl="//test/image.jpg" name="Alt Text" />);
    expect(screen.getByAltText("Alt Text")).toHaveAttribute(
      "src",
      "//test/image.jpg"
    );
  });

  /**
   * Préférons l'accessibilité par testing library pour tester nos composants
   */
  test.skip("renders correctly", () => {
    const { container } = render(
      <ProductImage imageUrl="//test/image.jpg" name="Alt Text" />
    );
    expect(container).toMatchSnapshot();
  });
});
