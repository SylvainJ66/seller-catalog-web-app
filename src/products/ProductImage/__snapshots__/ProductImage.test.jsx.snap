// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProductImage /> > renders correctly 1`] = `
<div>
  <img
    alt="Alt Text"
    src="//test/image.jpg"
  />
</div>
`;
